programs/mpl-staking/tests/test_claim.rs

```diff
@@ -259,517 +300,200 @@ async fn successeful_claim() -> Result<(), TransportError> {
 
-    let mint_governance = realm
-        .create_mint_governance(
-            context.mints[0].pubkey.unwrap(),
-            &context.mints[0].authority,
-            &voter,
-            voter_authority,
-            payer,
-            context
-                .addin
-                .update_voter_weight_record_instruction(&registrar, &voter),
-        )
-        .await;
-
-    let proposal = realm
-        .create_proposal(
-            mint_governance.address,
-            voter_authority,
-            &voter,
-            payer,
-            context
-                .addin
-                .update_voter_weight_record_instruction(&registrar, &voter),
-        )
-        .await
-        .unwrap();
-    realm
+    claim_setup
+        .realm
         .cast_vote(
-            mint_governance.address,
-            &proposal,
-            &voter,
-            voter_authority,
-            payer,
-            context
+            claim_setup.mint_governance.address,
+            &claim_setup.proposal,
+            &claim_setup.voter,
+            &claim_setup.voter_authority,
+            &claim_setup.payer,
+            claim_setup
+                .context
                 .addin
-                .update_voter_weight_record_instruction(&registrar, &voter),
+                .update_voter_weight_record_instruction(&claim_setup.registrar, &claim_setup.voter),
         )
         .await
         .unwrap();
     let vote_record = get_vote_record_address(
         &Pubkey::from(GOVERNANCE_PROGRAM_ID),
-        &proposal.address,
-        &proposal.owner_token_owner_record,
+        &claim_setup.proposal.address,
+        &claim_setup.proposal.owner_token_owner_record,
     );
 
-    context
+    claim_setup
+        .context
         .addin
-        .restrict_tokenflow(&registrar, &realm_authority, &voter)
+        .restrict_tokenflow(
+            &claim_setup.registrar,
+            &claim_setup.realm_authority,
+            &claim_setup.voter,
+        )
         .await
         .unwrap();
 
-    context
+    claim_setup
+        .context
         .addin
         .claim(
-            &rewards_pool,
-            reward_mint,
-            &deposit_mining,
-            voter_authority,
-            &voter_authority_ata,
-            &context.rewards.program_id,
-            &registrar,
-            &mint_governance.address,
-            &proposal.address,
-            &voter,
+            &claim_setup.rewards_pool,
+            &claim_setup.reward_mint,
+            &claim_setup.deposit_mining,
+            &claim_setup.voter_authority,
+            &claim_setup.voter_authority_ata,
+            &claim_setup.context.rewards.program_id,
+            &claim_setup.registrar,
+            &claim_setup.mint_governance.address,
+            &claim_setup.proposal.address,
+            &claim_setup.voter,
             &vote_record,
         )
         .await
-        .assert_on_chain_err(MplStakingError::TokenflowRestricted);
```

Comment:
afaik claiming will return errors from the `Staking`. i guess `rewards` even doesn't have such errors.